stock_delivery_result_import/wizards/stock_delivery_result_import.py

```diff
@@ -74,14 +76,16 @@ def import_stock_delivery_result(self):
             for picking, vals in pick_dict.items():
                 carrier_info = vals["carrier_info"]
                 tracking_refs = list(set(vals["tracking_refs"]))
+                if picking.carrier_tracking_ref:
+                    tracking_refs.append(picking.carrier_tracking_ref)
                 tracking_refs = ", ".join(ref for ref in tracking_refs)
                 picking.write(
-                    {
-                        "carrier_info_id": carrier_info and carrier_info.id or False,
-                        "carrier_tracking_ref": tracking_refs,
-                        "log_id": import_log.id,
-                    }
+                    {"carrier_tracking_ref": tracking_refs, "log_id": import_log.id}
                 )
+                if not picking.carrier_info_id:
+                    picking.write(
+                        {"carrier_info_id": carrier_info and carrier_info.id or False}
+                    )
```

Comment:
@TomohiroAbe-Quartile Avoid splitting the write operation on picking in two separate steps unless it's necessary.

Reply:
@yostashiro 
Sorry, I have no idea why unnecessary to splitting the write operation in this situation.